### DISCUSSION TITLE
Thank you!

### DISCUSSION THREAD

**Discussion by garreh89** [oatybiscuit/Node-RED-HA-Octopus-Agile-JSONata#4] [General]
Hi OatyBiscuit.

I've got to say this is an amazing piece of work! You've certainly put some time and effort into this.

Your instructions were so clear and spot on, I got this working in no time.
It's been running for 3 days now, and each time it's picked the best prices on Agile, they've been absolutely spot on . It's exactly what I've been looking for over the last 3 months.

Thank you for taking your time to develop this, everyone on Agile with Solar / Battery & HomeAssistant should know about this and be using it!

Kind regards,

**Comment by oatybiscuit** [comment]
Thank _you_ for your comments. I don't use Agile tariff myself, but this has been a very interesting exercise. The current code is about my third iteration over two years, and I am happy that it (still) works, and that it manages DST without issue. A great way to learn JSONata too.

I am most pleased with the DST local time management, and can confirm that it again coped with putting the clocks back yesterday. The 'best contiguous time periods' for export included:

27 Oct, 01:30 BST - 27 Oct, 01:30 GMT for 60 minutes @ 10.54 p

which, of course, it is!

**Comment by garreh89** [comment]
You've done a fantastic job, really appreciate the time, effort and learning that went into this.

To be honest, anyone who is using HA / Agile / Solar Battery Storage should be using this!

It makes the whole automation setup a lot easier to deal with and you don't have to login to HA every day to adjust thresholds when the prices fluctuate above your current automation fixed set points.

It worked absolutely flawlessly when DST changed, it kept everything exactly how it should have been.

Thanks again, really appreciate it! :)
